dashboard/app/models/eval.rb

```diff
@@ -89,4 +89,10 @@ def self.toolbox
   def toolbox(type)
     Eval.toolbox
   end
+
+  def embed_blocks(blocks)
+    match = /<xml><block type="functional_display".*?><functional_input.*?>(.*)<\/functional_input><\/block><\/xml>/.match(blocks)
+    return match[1] || blocks
+  end
```

Comment:
Since this regex isn't exactly self-explanatory, could you add a comment to this method describing what it's doing (and/or referencing the details in this PR)?